pkg/cortex/runtime_config.go

```diff
@@ -20,11 +21,13 @@ type runtimeConfigValues struct {
 }
 
 func loadRuntimeConfig(r io.Reader) (interface{}, error) {
-	var overrides = &runtimeConfigValues{}
+	content, err := ioutil.ReadAll(r)
+	if err != nil {
+		return nil, err
+	}
 
-	decoder := yaml.NewDecoder(r)
-	decoder.SetStrict(true)
-	if err := decoder.Decode(&overrides); err != nil {
+	var overrides = &runtimeConfigValues{}
+	if err := yaml.UnmarshalStrict(content, overrides); err != nil {
```

Comment:
I think we should revert back to original code, and extend the checks as follows:
- If first returned error is EOF, we return `nil`. This covers empty document case.
- We call `decoder.Decode` second time, and **if it doesn't return EOF**, then `loadRuntimeConfig` will return error.

Second check covers the case when there are multiple YAML documents on the input, eg.

```
---
# This is an empty YAML.
---
multi_kv_config:
  mirror_enabled: true
```

In this case, first two calls to `Decode` return `nil` (no error), so our current version of `loadRuntimeConfig` (or version suggested in this PR, which only decodes first document) would return no error, which is confusing.

WDYT?

Reply:
Very good point about multiple documents (that should be forbidden). I've changed it. What do you think?